control/controlhttp/client.go

```diff
@@ -517,10 +515,16 @@ func (a *Dialer) tryURLUpgrade(ctx context.Context, u *url.URL, addr netip.Addr,
 	// unexpected EOFs...), and we're bound to forget someday and
 	// introduce a protocol optimization at a higher level that starts
 	// eagerly transmitting from the server.
-	connCh := make(chan net.Conn, 1)
+	var lastConn atomic.Pointer[net.Conn]
```

Comment:
I'd use `syncs.AtomicValue[net.Conn]` here instead of `atomic.Pointer` (Pointer to interface is weird)

And our `syncs.AtomicValue` (unlike std's `atomic.Value`) handles concrete type changes.

Reply:
Nice, thanks! `syncs.AtomicValue` is exactly what I wanted here, but could not find.